src/otx/core/data/dataset/anomaly/dataset.py

```diff
@@ -0,0 +1,150 @@
+"""Anomaly Classification Dataset."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Callable
+
+import torch
+from anomalib.data.utils import masks_to_boxes
+from datumaro import DatasetSubset, Image
+from torchvision import io
+
+from otx.core.data.dataset.base import OTXDataset, Transforms
+from otx.core.data.entity.anomaly import (
+    AnomalyClassificationDataBatch,
+    AnomalyClassificationDataItem,
+    AnomalyDetectionDataBatch,
+    AnomalyDetectionDataItem,
+    AnomalySegmentationDataBatch,
+    AnomalySegmentationDataItem,
+)
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.mem_cache import NULL_MEM_CACHE_HANDLER, MemCacheHandlerBase
+from otx.core.types.image import ImageColorChannel
+from otx.core.types.task import OTXTaskType
+
+
+class AnomalyDataset(OTXDataset):
+    """OTXDataset class for anomaly classification task."""
+
+    def __init__(
+        self,
+        task_type: OTXTaskType,
+        dm_subset: DatasetSubset,
+        transforms: Transforms,
+        mem_cache_handler: MemCacheHandlerBase = NULL_MEM_CACHE_HANDLER,
+        mem_cache_img_max_size: tuple[int, int] | None = None,
+        max_refetch: int = 1000,
+        image_color_channel: ImageColorChannel = ImageColorChannel.RGB,
+        stack_images: bool = True,
+    ) -> None:
+        self.task_type = task_type
+        super().__init__(
+            dm_subset,
+            transforms,
+            mem_cache_handler,
+            mem_cache_img_max_size,
+            max_refetch,
+            image_color_channel,
+            stack_images,
+        )
+
+    def _get_item_impl(
+        self,
+        index: int,
+    ) -> AnomalyClassificationDataItem | AnomalySegmentationDataBatch | AnomalyDetectionDataBatch:
+        datumaro_item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
+        img = datumaro_item.media_as(Image)
+        # returns image in RGB format if self.image_color_channel is RGB
+        img_data, img_shape = self._get_img_data_and_shape(img)
+        # Note: This assumes that the dataset is in MVTec format.
+        # We can't use datumaro label id as it returns some number like 3 for good from which it is hard to infer
+        # whether the image is Anomalous or Normal. Because it leads to other questions like what do numbers 0,1,2 mean?
+        label: torch.LongTensor = (
+            torch.tensor(0.0, dtype=torch.long) if "good" in datumaro_item.id else torch.tensor(1.0, dtype=torch.long)
+        )
```

Comment:
@wonjuleee, it seems there are no binary case in `label_id` of Datumaro MVTec format.